processor/filterprocessor/filter_processor.go

```diff
@@ -73,34 +74,36 @@ func (fmp *filterMetricProcessor) GetCapabilities() component.ProcessorCapabilit
 }
 
 // Start is invoked during service startup.
-func (*filterMetricProcessor) Start(ctx context.Context, host component.Host) error {
+func (*filterMetricProcessor) Start(_ context.Context, _ component.Host) error {
 	return nil
 }
 
 // Shutdown is invoked during service shutdown.
-func (*filterMetricProcessor) Shutdown(ctx context.Context) error {
+func (*filterMetricProcessor) Shutdown(_ context.Context) error {
 	return nil
 }
 
 // ConsumeMetricsData implements the MetricsProcessor interface
-func (fmp *filterMetricProcessor) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
-	return fmp.next.ConsumeMetricsData(ctx, consumerdata.MetricsData{
-		Node:     md.Node,
-		Resource: md.Resource,
-		Metrics:  fmp.filterMetrics(md.Metrics),
-	})
+func (fmp *filterMetricProcessor) ConsumeMetrics(ctx context.Context, md pdata.Metrics) error {
+	return fmp.next.ConsumeMetrics(ctx, fmp.filterMetrics(md))
 }
 
 // filterMetrics filters the given spans based off the filterMetricProcessor's filters.
-func (fmp *filterMetricProcessor) filterMetrics(metrics []*metricspb.Metric) []*metricspb.Metric {
-	keep := make([]*metricspb.Metric, 0, len(metrics))
-	for _, m := range metrics {
-		if fmp.shouldKeepMetric(m) {
-			keep = append(keep, m)
+func (fmp *filterMetricProcessor) filterMetrics(md pdata.Metrics) pdata.Metrics {
+	mds := pdatautil.MetricsToMetricsData(md)
```

Comment:
Do we have an issue for updating `filtermetric.Matcher` to use the new data types as well so this translation back & forth isn't needed here?

Reply:
This is to unblock trace part to be implemented using the new format since trace part is stable but metrics is not yet. when metrics will be stable will change this.